include/camp/resource.hpp

```diff
@@ -172,18 +172,24 @@ namespace resources
 #endif
 
   template<typename Res>
-  struct EventProxy {
+  struct EventProxy : ::camp::resources::detail::EventProxyBase {
+    using native_event = decltype(std::declval<Res>().get_event());
+    
     EventProxy(EventProxy &&) = default;
-    EventProxy(EventProxy const &) = delete;
+    EventProxy(EventProxy &) = delete;
     EventProxy &operator=(EventProxy &&) = default;
-    EventProxy &operator=(EventProxy const &) = delete;
+    EventProxy &operator=(EventProxy &) = delete;
```

Comment:
why are these not const&?

Reply:
I can try to switch them to that, but the event being pulled out is not const, and the conversion methods were not const, so there were some issues there.  It really could use a pass to ensure const-correctness, which I will try to do but I'd love another set of eyes passing over EventProxy and Event for that if you have the time.

Reply:
I wasn't sure if you could delete the copy constructor, copy assignment operator unless it had the normal signature.

Reply:
I took a look over the Event related methods and didn't see anything out of place.

Reply:
Ok, I am still confused about the signature change so I read the cppreference copy constructor page. I thought the copy constructor had to be T(T const&,) but I see it can also be T(T&) so deleting either signature should work to delete the copy constructor. I don't see how the EventProxy copy constructor signature matters much here especially since its deleted.

Reply:
That makes sense, I'll try switching it back, and as you say that should actually behave just fine.  If it does it certainly makes more sense that way.

Reply:
Seems to behave just fine, so I reverted it.  Assuming tests pass, I'm planning to merge this version.